### ISSUE TITLE
Should services in forward mode parse headers?

### ISSUE THREAD

**Issue by danielkhan** [w3c/trace-context#251]
What is your opinion on sampling and it's relation to processing mode? 

Also there is a change that forward mode expects parsing headers now. Compare with existing spec:

![image](https://user-images.githubusercontent.com/9950081/50648771-e963f600-0f30-11e9-8b59-c88f76149fe6.png)

Do you think it is required?

_Originally posted by @SergeyKanzhelev in https://github.com/w3c/trace-context/pull/222#issuecomment-451195594_

**Comment by danielkhan** [comment]
Let's also use this issue to discuss the general behavior of forwarding systems wrt dropping invalid headers.

**Comment by danielkhan** [comment]
Proposed wording:

> A forwarding system MAY parse the headers and MAY drop them if they are invalid.

**Comment by AloisReitbauer** [comment]
LGTM

**Comment by yurishkuro** [comment]
+1

**Comment by SergeyKanzhelev** [comment]
I'd suggest to add a wording that implementation can choose to protect itself from headers that looks like a new version but carry a megabyte of text inside. 

```
02-traceid-spanid-01-<1 mg of data>
```

We either should decide how big the future version is allowed to be or let implementation decide this.

**Comment by danielkhan** [comment]
Agreed on the wording to make the behavior more concrete but the size limit needs more discussion and will be covered in a separate issue created by @SergeyKanzhelev.

**Comment by danielkhan** [comment]
Depends on #246. Wording will be changed, once the PR has landed or once it was decided to split it up.
